static/app/views/replays/detail/errorList.tsx

```diff
@@ -0,0 +1,190 @@
+import {Fragment, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import EventOrGroupExtraDetails from 'sentry/components/eventOrGroupExtraDetails';
+import EventOrGroupHeader from 'sentry/components/eventOrGroupHeader';
+import {PanelTable} from 'sentry/components/panels';
+import Placeholder from 'sentry/components/placeholder';
+import {DEFAULT_STREAM_GROUP_STATS_PERIOD} from 'sentry/components/stream/group';
+import GroupChart from 'sentry/components/stream/groupChart';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Group, NewQuery, PageFilters} from 'sentry/types';
+import DiscoverQuery from 'sentry/utils/discover/discoverQuery';
+import EventView from 'sentry/utils/discover/eventView';
+import theme from 'sentry/utils/theme';
+import useApi from 'sentry/utils/useApi';
+import {useLocation} from 'sentry/utils/useLocation';
+import useMedia from 'sentry/utils/useMedia';
+import useOrganization from 'sentry/utils/useOrganization';
+import withPageFilters from 'sentry/utils/withPageFilters';
+
+type Props = {
+  projectId: string;
+  replayId: string;
+  selection: PageFilters;
+};
+const columns = [t('Issue'), t('Graph'), t('Events'), t('Users')];
+
+function ErrorList(props: Props) {
+  const organization = useOrganization();
+  const location = useLocation();
+  const api = useApi();
+  const isScreenLarge = useMedia(`(min-width: ${theme.breakpoints[2]})`);
+
+  const [issues, setIssues] = useState<Group[]>([]);
+  const [issueStats, setIssuesStats] = useState<Group[]>([]);
+
+  const getEventView = () => {
+    const {selection} = props;
+    const eventQueryParams: NewQuery = {
+      id: '',
+      name: '',
+      version: 2,
+      fields: ['issue'],
+      environment: selection.environments,
+      projects: selection.projects,
+      query: `replayId:${props.replayId} AND event.type:error`,
+    };
+    const result = EventView.fromNewQueryWithLocation(eventQueryParams, location);
+    return result;
+  };
+
+  useEffect(() => {
+    const options = {
+      includeAllArgs: true,
+      query: {
+        collapse: 'stats',
+        expand: ['owner', 'inbox'],
+        project: props.projectId,
+        query: `replayId:${props.replayId}`,
+      },
+    };
+
+    api
+      .requestPromise(`/organizations/${organization.slug}/issues/`, options)
+      .then(async result => {
+        setIssues(result[0]);
+        const statsOptions = {
+          includeAllArgs: true,
+          query: {
+            project: props.projectId,
+            groups: result[0].map(issue => issue.id),
+            query: `replayId:${props.replayId}`,
+          },
+        };
+        const issuesResults = await api.requestPromise(
+          `/organizations/${organization.slug}/issues-stats/`,
+          statsOptions
+        );
+        setIssuesStats(issuesResults[0]);
+      });
+  }, [api, organization.slug, props.replayId, props.projectId]);
```

Comment:
I had two useEffects before and refactored this into one since the 2nd call depends on the group ids from the first. I'm all for making this better if anyone has suggestions.

Reply:
#### 1/3
A single `useEffect` call looks right to me. If there was a 2nd `useEffect` it would run but be a no-op when the component first renders, then it would run again after the 1st request comes back. So you'd need to add the if statement to handle that no-op case ... one useEffect is better 👍 .

---

#### 2/3
Style-wise we've got a mix of `.then()` and `await` in here which is something to deal with. The way to use `await` everywhere is to 1) have an external async method 2) and call that from inside the useEffect anon function 3) don't await the results.

```
const fetchIssueData = useCallback(async () => {
  const issues = await api.requestPromise(`/organizations/${organization.slug}/issues/`, { ... });
  setIssues(issues[0]);
  const issueStats = await api.requestPromise(`/organizations/${organization.slug}/issues-stats/`, { ... });
  setIssuesStats(issueStats[0]);
}, [api, organization.slug, props.replayId, props.projectId]);

useEffect(() => {
  fetchIssueData();
}, [fetchIssueData]);
```

--- 

#### 3/3
Once that style stuff is done it's easier to notice that there's no error handing happening. If the requests fail then errors will be thrown without anything to catch them. So `fetchIssueData` would need to have try/catch block(s) inside it and set some kind of error state.

I'm fine with error handling being a followup PR too

Reply:
I made the function and went ahead and added error handling, it felt weird to do 90% of the work. Added a screenshot of the placeholders.